src/main/java/io/knotx/junit5/wiremock/KnotxWiremockExtension.java

```diff
@@ -101,31 +102,52 @@ public boolean supportsParameter(
             "Annotating String with KnotxWiremock is not supported");
       }
     }
-
     return false;
   }
 
   @Override
   public Object resolveParameter(
       ParameterContext parameterContext, ExtensionContext extensionContext)
       throws ParameterResolutionException {
-    Class<?> type = getType(parameterContext);
 
-    if (type.equals(WireMockServer.class) || type.equals(Integer.class)) {
-      KnotxWiremock knotxWiremock = ReflectUtil.getWiremockAnnotation(parameterContext);
-
-      String name = getClassFieldName(extensionContext, parameterContext);
+    return parameterContext
+        .findAnnotation(KnotxWiremock.class).map(knotxWiremock -> {
+          String nameReference = getFullyQualifiedName(extensionContext, parameterContext,
```

Comment:
This may break some promises made in docs (i.e. that server with given parameter name will be reusable for the whole test class), please review them and change accordingly if this is the intended behavior.

Reply:
You are right. The situation will happen when you specify static port in parameter. In other cases it works as in documentation. There is Unit test for that.